pkg/readiness/ready_tracker.go

```diff
@@ -581,6 +623,15 @@ func (t *Tracker) statsPrinter(ctx context.Context) {
 				log.Info("unsatisfied data", "name", u.namespacedName, "gvk", u.gvk)
 			}
 		}
+		logUnsatisfiedAssignMetadata(t)
+	}
+}
+
+func logUnsatisfiedAssignMetadata(t *Tracker) {
+	defer t.assignMetadata.mu.RUnlock()
+	t.assignMetadata.mu.RLock()
+	for amKey := range t.assignMetadata.expect {
```

Comment:
This should call `assignMetadata.unsatisfied()` rather than reaching into the internals of the tracker. It shouldn't cross the boundary and lock the objectTracker's mutex - it should let the objectTracker synchronize access to its own fields. In retrospect the `objectTracker` could be in a separate package which would enforce this restriction.

Reply:
Done